src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,80 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Node<K, V>[] table = new Node[DEFAULT_CAPACITY];
+    private int size;
 
     @Override
     public void put(K key, V value) {
-
+        final Node<K, V> newNode = new Node<>(key, value, null);
+        int index = getIndex(key);
+        Node<K, V> currentNode = table[index];
+        while (currentNode != null) {
+            if (currentNode.key == newNode.key || currentNode.key != null
+                    && currentNode.key.equals(newNode.key)) {
+                currentNode.value = newNode.value;
+                break;
+            } else if (currentNode.next == null) {
+                currentNode.next = newNode;
+                size++;
+                break;
+            }
+            currentNode = currentNode.next;
+        }
+        if (currentNode == null) {
+            table[index] = newNode;
+            size++;
+        }
+        resize();
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> node = table[getIndex(key)];
+        while (node != null) {
+            if (node.key == key || node.key != null && node.key.equals(key)) {
+                return node.value;
+            } else {
+                node = node.next;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(K key, V value, Node<K, V> next) {
```

Comment:
null is always passed here
```suggestion
        Node(K key, V value) {
```